libs/wingc/src/jsify.rs

```diff
@@ -896,6 +897,15 @@ impl<'a> JSifier<'a> {
 					if_let_value,
 					self.jsify_expression(value, ctx)
 				));
+				let elif_let_value = "$ELIF_LET_VALUE";
+				for i in 0..elif_statements.len() {
+					let value = format!("{}{}", elif_let_value, i);
+					code.line(format!(
```

Comment:
I think it's wrong to evaluate all the values in advance. Consider:
```wing
if let x = tryA() {
} elif let x = tryB() {
} elif let x = TryC() {
}
```
1. There's a performance penalty evaluating `TryA`, `TryB` and `TryC` in advance if `TryA` succeeds.
2. If there are side effects to the evaluation then the use wouldn't expect them to happen if `TryA` succeeds.

Reply:
From what I understand, this code in Wing:
```ts (wing)
if let x = tryA() {
   ...
} elif let x = tryB() {
   ...
} elif let x = TryC() {
   ...
}
```
will generate this code in TypeScript:
```ts
let $IF_LET_VALUE = tryA();
let $ELIF_LET_VALUE0 = tryB();
let $ELIF_LET_VALUE1 = tryC();
if ($IF_LET_VALUE !== undefined) {
   ...
} else if ($ELIF_LET_VALUE0 !== undefined) {
   ...
} else if ($ELIF_LET_VALUE1 !== undefined) {
   ...
}
```
In order for the evaluation of the `elif` conditions to only occur in the case of the else, you can use the following code:
```ts
let $IF_LET_VALUE = tryA();
if ($IF_LET_VALUE !== undefined) {
   ...
} else {
   let $ELIF_LET_VALUE0 = tryB();
   if ($ELIF_LET_VALUE0 !== undefined) {
      ...
   } else {
      let $ELIF_LET_VALUE1 = tryC();
      if ($ELIF_LET_VALUE1 !== undefined) {
         ...
      }
   }
}
```
This code might be considered less elegant, but it's effective in this case.